### PR TITLE
Limits the number of scopes rows in the listing

### LINKED ISSUE #22: Limit the number of scopes rows in the listing
At the moment, the number of scope in the listing is infinite.

image

It could be great to limit to 2 rows (about 5/6 scopes)

### PR DESCRIPTION
**Pull request by MaximeBro**
- Fixes #22 & #28 

Capture d’écran 2024-10-08 à 10 10 23

Capture d’écran 2024-10-09 à 14 11 05

### CODE REVIEW

File: src/app/issues/PageIssues.tsx [hunk 1]
```diff
@@ -177,6 +178,8 @@ export const PageIssues = () => {
 
   const styles = useFieldSelectScopeStyles();
 
+  console.log({ issues });
```
Comment (yoannfleurydev) [inline src/app/issues/PageIssues.tsx:181]:
```suggestion
```

File: src/app/issues/PageIssues.tsx [hunk 2]
```diff
@@ -488,48 +496,191 @@ export const PageIssues = () => {
...
+                                                )[700],
+                                              }}
+                                            >
+                                              {scope.name}
+                                            </Tag>
+                                          ))}
+                                      </Wrap>
+                                    }
+                                  >
+                                    <Tag style={{ zIndex: 9999 }}>{`+${
```
Comment (HugoPerard) [inline src/app/issues/PageIssues.tsx:581]: I think it's not normal to do that 😅
And for you knowledge, on Chakra UI components you a zIndex props, instead on use style -> zIndex

File: src/app/issues/PageIssues.tsx [hunk 3]
```diff
@@ -488,48 +496,191 @@ export const PageIssues = () => {
...
+                                      issue.scopes.length - 1
+                                    }`}</Tag>
```
Comment (HugoPerard) [inline src/app/issues/PageIssues.tsx:583]: We don't need to convert as number (with "+" to put it back at string after (with literal template between `)

File: src/app/issues/PageIssues.tsx [hunk 4]
```diff
@@ -488,48 +496,191 @@ export const PageIssues = () => {
...
+                                      <Wrap>
+                                        {issue.scopes
+                                          ?.slice(1, issue.scopes.length)
+                                          .map(({ scope }) => (
+                                            <Tag
+                                              key={scope.id}
+                                              color={
+                                                generateSwatch(
+                                                  scope.color ?? brandColor
+                                                )[700]
```
Comment (HugoPerard) [inline src/app/issues/PageIssues.tsx:559]: Two things :

1. Here you calculate every time a new swatch from the same color, it's better to declare one time the `generateSwatch(scope.color ?? brandColor)` and then use the result each time we need a color tint
2. In this case, we should not defined manually color, bg, dark stuff... You can use colorScheme props instead

File: src/app/issues/PageIssues.tsx [hunk 5]
```diff
@@ -488,48 +496,191 @@ export const PageIssues = () => {
...
+                                          >
+                                            {scope.name}
+                                          </Tag>
+                                        ))}
+                                    </Wrap>
+                                  }
+                                >
+                                  <Tag style={{ zIndex: 9999 }}>{`+${
+                                    issue.scopes.length - 5
+                                  }`}</Tag>
```
Comment (HugoPerard) [inline src/app/issues/PageIssues.tsx:678]: Same remark than above

File: src/app/issues/PageIssues.tsx [hunk 6]
```diff
@@ -488,48 +496,191 @@ export const PageIssues = () => {
...
+                                                scope.color ?? brandColor
+                                              )[100]
+                                            }
+                                            _dark={{
+                                              color: generateSwatch(
+                                                scope.color ?? brandColor
+                                              )[50],
+                                              bg: generateSwatch(
+                                                scope.color ?? brandColor
+                                              )[700],
```
Comment (HugoPerard) [inline src/app/issues/PageIssues.tsx:667]: Same remark than above

File: src/app/issues/PageIssues.tsx [hunk 7]
```diff
@@ -488,48 +496,191 @@ export const PageIssues = () => {
...
+                                  {scope.name}
+                                </Tag>
+                              ))}
+
+                              {issue.scopes.length > 5 ? (
+                                <Tooltip
+                                  hasArrow
+                                  p={2}
+                                  borderRadius={8}
+                                  bg="gray.800"
```
Comment (HugoPerard) [inline src/app/issues/PageIssues.tsx:642]: Did you try in light mode ?

**Review by HugoPerard — CHANGES_REQUESTED** [review]
Can you add updated screenshot please

**Comment by HugoPerard** [comment]
@MaximeBro On mobile, we want text to be align on the left, not centered. That will also fix the alignment of the items according to the title/description length

image

File: src/app/issues/PageIssues.tsx [hunk 8]
```diff
@@ -60,6 +62,7 @@ import { SearchInput } from '@/components/SearchInput';
 import { generateSwatch } from '@/utils/colors';
```
Comment (yoannfleurydev) [inline src/app/issues/PageIssues.tsx:62]: Remove this import as it is not used anymore in that file.

**Review by yoannfleurydev — CHANGES_REQUESTED** [review]
image

please fix the annotation on that PR

**Review by yoannfleurydev — APPROVED** [review]

**Pull request merged**